### PR TITLE
[Bindless][CUDA] Unique addressing modes per dimension

### PR DESCRIPTION
**Pull request by Seanst98**
Extends the CUDA adapter to allow for unique addressing modes per dimension

Modifies:
- `image.cpp`
- `sampler.hpp`
- `sampler.cpp`

Corresponding intel-llvm PR: https://github.com/intel/llvm/pull/12109

### CODE REVIEW

**Review by isaacault — APPROVED** [review]

**Comment by Seanst98** [comment]
> Why are modified files listed in the commit message?

I've removed them from the commit message.

**Comment by Seanst98** [comment]
Pinging @oneapi-src/unified-runtime-cuda-write  for approvals/reviews and ready to merge tag.

File: source/adapters/cuda/image.cpp [hunk 1]
```diff
@@ -239,29 +239,38 @@ ur_result_t urTextureCreate(ur_sampler_handle_t hSampler,
...
-      AddrMode = CU_TR_ADDRESS_MODE_WRAP;
-    } else if (AddrModeProp == (UR_SAMPLER_ADDRESSING_MODE_MIRRORED_REPEAT -
-                                UR_SAMPLER_ADDRESSING_MODE_NONE)) {
-      AddrMode = CU_TR_ADDRESS_MODE_MIRROR;
+    CUaddress_mode AddrMode[3];
+    for (size_t i = 0; i < 3; i++) {
+      ur_sampler_addressing_mode_t AddrModeProp =
+          hSampler->getAddressingModeDim(i);
+      if (AddrModeProp == (UR_SAMPLER_ADDRESSING_MODE_CLAMP_TO_EDGE -
+                           UR_SAMPLER_ADDRESSING_MODE_NONE)) {
```
Comment (GeorgeWeb) [inline source/adapters/cuda/image.cpp:260]: Do we need to always substract `UR_SAMPLER_ADDRESSING_MODE_NONE` which is 0?

**Review by GeorgeWeb — APPROVED** [review]
CUDA changes LGTM after connecting it with the changes from intel/llvm.

.. a minor comment but the changes look fine.

File: source/adapters/cuda/image.cpp (hunk 1 above)
Comment (Seanst98) [reply source/adapters/cuda/image.cpp:260]: That's a good point. It's a carry-over from before UR where `PI_SAMPLER_ADDRESSING_MODE_NONE` was not 0.

Would you prefer I remove it?

**Pull request merged**